compiler/rustc_mir_build/src/builder/matches/test.rs

```diff
@@ -175,6 +175,30 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                         place = ref_str;
                         ty = ref_str_ty;
                     }
+                    &ty::Pat(base, _) => {
+                        assert_eq!(ty, value.ty());
+
+                        let transmuted_place = self.temp(base, test.span);
+                        self.cfg.push_assign(
+                            block,
+                            self.source_info(scrutinee_span),
+                            transmuted_place,
+                            Rvalue::Cast(CastKind::Transmute, Operand::Copy(place), base),
```

Comment:
Can you add a comment explaining that we only support pattern types on the basic scalar types rn which all implement `Copy` so its fine to just unconditionally do a `Operand::Copy` here instead of the usual "is it copy mod regions" thing.